src/nvidia_rag/rag_server/server.py

```
@@ -1173,6 +1617,322 @@ def sanitize_query_for_logging(query):
+        return JSONResponse(content={"message": str(e)}, status_code=status_code)
+    except ValueError as e:
+        # Handle validation errors
+        logger.warning(
+            "Validation error in /vector_stores/{vector_store_id}/search endpoint: %s",
+            e,
+        )
+        return JSONResponse(
+            content={"message": str(e)},
+            status_code=ErrorCodeMapping.BAD_REQUEST,
+        )
+    except Exception as e:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)